packages/blade/src/components/SideNav/SideNavItems/SideNavLink.tsx

```diff
@@ -0,0 +1,327 @@
+        </BaseBox>
+      ) : null}
+      <BaseText
+        truncateAfterLines={1}
+        color="currentColor"
+        fontWeight="medium"
+        fontSize={100}
+        lineHeight={100}
+        as="p"
+        className={isL1Item ? HIDE_WHEN_COLLAPSED : ''}
+      >
+        {title}
+      </BaseText>
+      {titleSuffix ? <BaseBox>{titleSuffix}</BaseBox> : null}
+    </Box>
+  );
+};
+
+const L3Trigger = ({
+  title,
+  icon,
+  as,
+  href,
+  titleSuffix,
+  tooltip,
+}: Pick<
+  SideNavLinkProps,
+  'title' | 'icon' | 'as' | 'href' | 'titleSuffix' | 'tooltip'
+>): React.ReactElement => {
+  const { onExpandChange, isExpanded, collapsibleBodyId } = useCollapsible();
+
+  const toggleCollapse = (): void => onExpandChange(!isExpanded);
+  const iconProps = {
+    size: 'medium',
+    color: 'currentColor',
+  } as const;
+
+  return (
+    <TooltipifyNavLink tooltip={tooltip}>
+      <StyledNavLinkContainer
+        as={href ? as : 'button'}
+        to={href}
+        onClick={toggleCollapse}
+        {...makeAccessible({ expanded: isExpanded, controls: collapsibleBodyId })}
+      >
+        <NavLinkIconTitle title={title} icon={icon} isL1Item={false} titleSuffix={titleSuffix} />
+        <BaseBox display="flex" alignItems="center">
+          {isExpanded ? <ChevronUpIcon {...iconProps} /> : <ChevronDownIcon {...iconProps} />}
+        </BaseBox>
+      </StyledNavLinkContainer>
+    </TooltipifyNavLink>
+  );
+};
+
+const CurvedVerticalLine = styled(BaseBox)((props) => {
+  const { colors, border, spacing } = props.theme;
+  return {
+    borderWidth: makeBorderSize(props.theme.border.width.thin),
+    borderColor: `${colors.transparent} ${colors.transparent} ${colors.surface.border.primary.muted} ${colors.surface.border.primary.muted}`,
+    borderStyle: 'solid',
+    borderRadius: `${makeBorderSize(border.radius.none)} ${makeBorderSize(
+      border.radius.none,
+    )} ${makeBorderSize(border.radius.none)} ${makeBorderSize(border.radius.medium)}`,
+    height: '100vh',
+    position: 'absolute',
+    top: `calc(-100vh + ${makeSize(NAV_ITEM_HEIGHT / 2)})`,
+    width: makeSpace(spacing[3]),
+    left: makeSpace(-spacing[3]),
+  };
+});
+
+const SideNavLink = ({
+  title,
+  href,
+  children,
+  titleSuffix,
+  trailing,
+  isActive,
+  icon,
+  tooltip,
+  as,
+}: SideNavLinkProps): React.ReactElement => {
+  const {
+    l2PortalContainerRef,
+    onLinkActiveChange,
+    closeMobileNav,
+    isL1Collapsed,
+    setIsL1Collapsed,
+  } = useSideNav();
+  const { level: _prevLevel } = useNavLink();
+  const prevLevel = _prevLevel ?? 0;
+  const currentLevel = prevLevel + 1;
+  const isL2Trigger = Boolean(children) && currentLevel === 1;
+  const isL3Trigger = Boolean(children) && currentLevel === 2;
+
+  console.count('SideNavLink');
+
+  const isFirstRender = useFirstRender();
+
+  const { refs, context } = useFloating({
+    open: isActive,
+  });
+
+  React.useLayoutEffect(() => {
```

Comment:
We can avoid useEffect by putting this code in onOpenChange inside useFloating.

Reply:
We do anyway have to keep useLayoutEffect to pass the initial state's value in the start. Better to keep it at one place than two